mittab/libs/assign_rooms.py

```diff
@@ -0,0 +1,64 @@
+import random
+from django.db import transaction
+from mittab.apps.tab.models import RoomCheckIn, Round, TabSettings
+from mittab.libs import errors, mwmatching, tab_logic
+
+
+def add_rooms():
+    no_seeding = TabSettings.get("disable_room_seeding", 0)
+
+    # Clear any existing room assignments
+    Round.objects.filter(round_number=TabSettings.get(
+        "cur_round") - 1).update(room=None)
+    round_number = TabSettings.get("cur_round") - 1
+
+    rooms = RoomCheckIn.objects.filter(
+        round_number=round_number).select_related("room")
+    rooms = sorted((r.room for r in rooms), key=lambda r: r.rank, reverse=True)
+    pairings = tab_logic.sorted_pairings(round_number)
+
+    if no_seeding:
+        random.shuffle(pairings)
+
+    if not pairings or not rooms or len(pairings) > len(rooms):
+        raise errors.RoomAssignmentError("Not enough rooms or pairings")
+
+    graph_edges = []
+
+    for pairing_i, pairing in enumerate(pairings):
+        for room_i, room in enumerate(rooms):
+            weight = 0
+
+            # High seed high room bonus
+            if no_seeding == 0:
```

Comment:
let's move the == 0 check to where we instantiate no_seeding, it's more intuitive to make that a bool right away